sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
-        private final Long offset;
-        private final String partitionKey;
-        private final Instant enqueuedTime;
-        private final Long sequenceNumber;
+    public String getReplyTo() {
+        String replyTo = null;
+        AmqpAddress amqpAddress = amqpAnnotatedMessage.getProperties().getReplyTo();
+        if (amqpAddress != null) {
+            replyTo = amqpAddress.toString();
+        }
+        return replyTo;
+    }
 
-        SystemProperties() {
-            super();
-            offset = null;
-            partitionKey = null;
-            enqueuedTime = null;
-            sequenceNumber = null;
+    /**
+     * Sets the address of an entity to send replies to.
+     *
+     * @param replyTo ReplyTo property value of this message
+     *
+     * @return The updated {@link EventData}.
+     * @see #getReplyTo()
+     */
+    public EventData setReplyTo(String replyTo) {
+        AmqpAddress replyToAddress = null;
+        if (replyTo != null) {
+            replyToAddress = new AmqpAddress(replyTo);
         }
+        amqpAnnotatedMessage.getProperties().setReplyTo(replyToAddress);
+        return this;
+    }
+
+    /**
+     * Gets the duration before this message expires.
+     * <p>
+     * This value is the relative duration after which the message expires, starting from the instant the message has
+     * been accepted and stored by the broker, as captured in {@link #getScheduledEnqueueTime()}. When not set
+     * explicitly, the assumed value is the DefaultTimeToLive set for the respective queue or topic. A message-level
+     * TimeToLive value cannot be longer than the entity's DefaultTimeToLive setting and it is silently adjusted if it
+     * does.
+     *
+     * @return Time to live duration of this message
+     */
+    public Duration getTimeToLive() {
+        return amqpAnnotatedMessage.getHeader().getTimeToLive();
+    }
+
+    /**
+     * Sets the duration of time before this message expires.
+     *
+     * @param timeToLive Time to Live duration of this message
+     *
+     * @return The updated {@link EventData}.
+     * @see #getTimeToLive()
+     */
+    public EventData setTimeToLive(Duration timeToLive) {
+        amqpAnnotatedMessage.getHeader().setTimeToLive(timeToLive);
+        return this;
+    }
+
+    /**
+     * Gets the session identifier for a session-aware entity.
+     *
+     * <p>
+     * For session-aware entities, this application-defined value specifies the session affiliation of the message.
+     * Messages with the same session identifier are subject to summary locking and enable exact in-order processing and
+     * demultiplexing. For session-unaware entities, this value is ignored.
+     * </p>
+     *
+     * @return The session id of the {@link EventData}.
+     * @see <a href="https://docs.microsoft.com/azure/service-bus-messaging/message-sessions">Message Sessions</a>
+     */
+    public String getSessionId() {
```

Comment:
EH has no session id

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version